aws/src/main/java/org/apache/iceberg/aws/AwsClientProperties.java

```diff
@@ -240,6 +248,22 @@ public <T extends AwsClientBuilder> void applyRetryConfigurations(T builder) {
     builder.overrideConfiguration(configBuilder.retryStrategy(RetryMode.ADAPTIVE_V2).build());
   }
 
+  /**
+   * Add a legacy md5 plugin if it is enabled.
+   *
+   * <p>Sample usage:
+   *
+   * <pre>
+   *     S3Client.builder().applyMutation(awsClientProperties::applyLegacyMd5Plugin)
+   * </pre>
+   */
+  @SuppressWarnings("rawtypes")
```

Comment:
is this suppress needed?

Reply:
I am not against the annotation if it is needed. I just wasn't sure which code caused this warning.

Reply:
Can we update the 3 apple methods to generics like this?
```
public <BuilderT extends AwsClientBuilder<BuilderT, ClientT>, ClientT> void applyLegacyMd5Plugin(BuilderT builder)
```

that should fix the raw type warning.